ios/Trifle/Sources/Trifle.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  Trifle.swift
+//  Trifle
+//
+
+import Foundation
+
+public class Trifle {
+    
+    public static let version = "0.1.3"
+    
+    private let contentSigner: ContentSigner
+    
+    public init(
+        tag: String,
+        with level: Level = .secureEnclave
+    ) {
+        switch level {
+        case .secureEnclave:
+            self.contentSigner =
+            SecureEnclaveDigitalSignatureKeyManager(tag: tag)
+        }
+    }
+    
+    /**
+     Create a new mobile Trifle keypair for which can be used to create a
+     certificate request and to sign messages. The library (Trifle) will
+     automatically try to choose the best algorithm and key type available on
+     this device.
+
+     - returns: An opaque Trifle representation, `KeyHandle`,
+        of the key-pair, which the client will need to store.
+     */
+    public func generateSigningKeyHandle() {
+        // TODO: IMPLEMENT
+    }
+    
+    /**
+     Generate a Trifle MobileCertificateRequest, signed by the provided
+     keyHandle, that can be presented to the Certificate Authority (CA) for
+     verification.
+     
+     - returns: An opaque Trifle representation
+        `MobileCertificateRequest` of the certificate request.
+     */
+    public func generateMobileCertificateRequest(
+        format: CSRFormat,
+        version: UInt32 = 0
+    ) throws -> MobileCertificateRequest {
+        switch format {
+        case let .PKCS10(x500Names, attributes):
+            let csr = try PKCS10CertificationRequest.Builder()
+                .addNames(x500Names)
+                .addAttributes(attributes)
+                .sign(with: contentSigner)
+            
+            return MobileCertificateRequest(
+                version: version,
+                pkcs10_request: Data(csr.octets)
+            )
+        }
+    }
+    
+    /**
+     // TODO(dcashman): define message format //
+     Sign the provided data with the provided key, including appropriate Trifle
+     metadata, such as the accompanying certificate.
+
+     - parameters: data - raw data to be signed.
+     - parameters: keyHandle - key to use for the signing. Must match the key used to
+        generate the Certificate.
+     - parameters: certificate - certificate to be included in the SignedData message.
+        Must match the key in keyHandle.
+
+     - returns:`SignedData` - signed data message in the Trifle format.
+    */
+    public func createSignedData(
+        data: Data,
+        certificate: Certificate
```

Comment:
missing keyhandle?

Reply:
Type does not exist yet. This PR just sets up a common file to work with

Reply:
added a placeholder